filebeat/crawler/prospector_log.go

```diff
@@ -140,6 +140,9 @@ func (p *ProspectorLog) scanGlob(glob string) {
 		// Track the stat data for this file for later comparison to check for
 		// rotation/etc
 		p.prospectorList[h.Path] = *h.Stat
+		// TODO: Registrar now updated also if no changes happened. Improve efficiency here
+		// Update registrar
+		p.Prospector.registrar.Persist <- h.GetState()
```

Comment:
Isn't the initial state set by LoadState? In case you are right and harvester and prospector would race for a state, worst case would be that an older state would be set than it actually is, but is overwritten with the correct state the next time state is written. That would be, in case filebeat is shut down at this moment, worst case would be to have some duplicated log lines?